pkg/arvo/app/publish.hoon

```diff
@@ -955,14 +988,16 @@
 ++  allowed
   |=  [who=@p mod=?(%read %write) book=@tas]
   ^-  ?
-  =/  scry-bek  /(scot %p our.bol)/permission-store/(scot %da now.bol)
   =/  book=notebook  (~(got by books) our.bol book)
-  =/  scry-pax
-    ?:  =(%read mod)
-      subscribers.book
-    writers.book
-  =/  full-pax  :(weld scry-bek /permitted/(scot %p who) scry-pax /noun)
-  .^(? %gx full-pax)
+  =/  =group-id
+    (need (group-id:de-path:group-store writers.book))
+  =/  role=(unit role-tag)
+    (role-for-ship:grup group-id who)
+  ?~  role
+    %.n
+  ?.  ?=(%write mod)
+    %.y
+  !=(%member u.role)
```

Comment:
A downside of doing this with role tags is that you can't have different notebooks in a group with different posting permissions.

Reply:
revamped with per-notebook role tags